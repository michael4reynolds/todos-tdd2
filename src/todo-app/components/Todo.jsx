import React from 'react'
import classNames from 'classnames'
import '../styles/styles.css'

const Todo = ({onClick, text, completed}) => {
  let liClass = classNames({completed})
  return (
    <li onClick={onClick} className={liClass}>{text}</li>
  )
}

export default Todo
