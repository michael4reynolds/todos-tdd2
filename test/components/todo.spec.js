import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import Todo from '../../src/todo-app/components/todo'

describe('App', () => {
  it('renders a list element', () => {
    const todo = shallow(<Todo/>)
    expect(todo.type()).toBe('li')
  })

  it('handles click event', () => {
    const onClick = jest.fn()
    const todo = shallow(<Todo onClick={onClick}/>)
    todo.simulate('click')
    expect(onClick).toBeCalled()
  })

  it('renders given text', () => {
    const todo = shallow(<Todo text="todo"/>)
    expect(todo.render().text()).toEqual('todo')
  })

  it('is not crossed when not completed', () => {
    const todo = shallow(<Todo completed={false}/>)
    expect(todo.props().style.textDecoration).toBe('none')
    expect(toJson(todo)).toMatchSnapshot()
  })

  it('is crossed when completed', () => {
    const todo = shallow(<Todo completed={true}/>)
    expect(todo.props().style.textDecoration).toBe('line-through')
    expect(toJson(todo)).toMatchSnapshot()
  })
})
