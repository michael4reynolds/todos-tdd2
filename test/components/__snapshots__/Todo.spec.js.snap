exports[`App is crossed when completed 1`] = `
<li
  style={
    Object {
      "textDecoration": "line-through",
    }
  } />
`;

exports[`App is not crossed when not completed 1`] = `
<li
  style={
    Object {
      "textDecoration": "none",
    }
  } />
`;
